ast/src/parsed/display.rs

```diff
@@ -679,7 +683,7 @@ impl<Ref: Display> Display for Expression<Ref> {
             Expression::Tuple(items) => write!(f, "({})", format_list(items)),
             Expression::LambdaExpression(lambda) => write!(f, "{lambda}"),
             Expression::ArrayLiteral(array) => write!(f, "{array}"),
-            Expression::BinaryOperation(left, op, right) => {
+            Expression::BinaryOperation(BinaryOperation { left, op, right }) => {
                 Expression::format_binary_operation(left, op, right, f)
```

Comment:
We should implement Display for BinaryOperation and "call" it here.

Reply:
I think it's ready. I had to rearrange the new `precedence` function/trait a bit. When we move on to the PR of `unary operations`, I'll finish moving the remaining function (`format_unary_operation`). `Main` has already been merged, so the `Number` struct is already included in this PR as well.

Let me know what you think :)